client/src/panels/result_graphs/result_flat_table.js

```diff
@@ -0,0 +1,268 @@
+    })
+      .then( () => this.setState({loading: false}) );
+  }
+  render(){
+    const { 
+      subject,
+      subject_result_counts,
+      drr_doc,
+    } = this.props;
+    const { loading, status_active_list } = this.state;
+    
+    if (loading) {
+      return (
+        <div style={{position: "relative", height: "80px", marginBottom: "-10px"}}>
+          <SpinnerWrapper config_name={"tabbed_content"} />
+        </div>
+      );
+    } else {
+      const flat_indicators = get_indicators(subject, drr_doc);
+      const icon_counts = _.countBy(flat_indicators, ({indicator}) => indicator.status_key);
+      const filtered_indicators = _.filter(flat_indicators, ind => _.isEmpty(status_active_list) || _.includes(status_active_list,ind.indicator.status_key));
+      const toggle_status_status_key = (status_key) => this.setState({status_active_list: _.toggle_list(status_active_list, status_key)});
+      const clear_status_filter = () => this.setState({status_active_list: []});
+      
+      return (
+        <div>
+          <div className="medium_panel_text">
+            <TM k="result_flat_table_text" args={{subject, ...subject_result_counts}}/>
+          </div>
+          <HeightClippedGraph clipHeight={700}>
+            <div style={{padding: '10px 10px'}}>
+              <StatusIconTable 
+                active_list={status_active_list}
+                icon_counts={icon_counts} 
+                onIconClick={toggle_status_status_key}
+                onClearClick={clear_status_filter}
+              />
+            </div>
+            <div className="results-flat-table">
+              {indicator_table_from_list(filtered_indicators, !subject.is_first_wave && drr_doc === 'drr17')}
+            </div>
+          </HeightClippedGraph>
+        </div>
+      );
+    }
+
+  }
+}
+
+export const declare_results_table_panel = () => declare_panel({
+  panel_key: "results_flat_table",
+  levels: ["dept", "crso", "program"],
+  panel_config_func: (level, panel_key) => ({
+    footnotes: false,
+    depends_on: ["programSpending", "programFtes"],
+    source: (subject) => get_source_links(["DP","DRR"]),
+    requires_result_counts: level === 'dept',
+    requires_granular_result_counts: level !== 'dept',
+    calculate(subject){
+      const subject_result_counts = level === 'dept' ?
+        ResultCounts.get_dept_counts(subject.id) :
+        GranularResultCounts.get_subject_counts(subject.id);
+
+      const had_doc_data = (doc) => {
+        const count_key = `${doc}_total`;
+        return (
+          /drr/.test(doc) &&
+          !_.isUndefined(subject_result_counts) && 
+          !_.isNull(subject_result_counts[count_key]) && 
+          subject_result_counts[count_key] > 0
+        );
+      };
+
+      const docs_with_data = _.chain(result_docs)
+        .keys()
+        .filter( had_doc_data )
+        .value();
+
+      const drr_doc = _.chain(docs_with_data)
+        .filter(doc => /drr/.test(doc))
```

Comment:
This test is already done on line 218, right? Maybe I'm confused about what's going on here.

Reply:
Nope, you're right, that line is extraneous. I also renamed `drr_doc` to `last_drr_doc` to make it clear what the different variables are for.